### DISCUSSION TITLE
slicing point cloud

### DISCUSSION THREAD

**Discussion by lurunawjan** [isl-org/Open3D#6343] [General]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).

### My Question

Hello, I wanted to ask, is there a way to slice point cloud either in horizontally or vertically? 

Thank you

**Comment by TheMadRey** [comment]
The way i'm doing this is by down sampling the original_pcd and then also create a voxelgrid from the original_pcd.
Then i iterate over all the voxels and check in what row they are.

```
import open3d as o3d
import numpy as np

voxel_size = 5

pcd_down = pcd.voxel_down_sample(voxel_size=voxel_size)
voxel_grid = o3d.geometry.VoxelGrid.create_from_point_cloud(pcd, voxel_size=voxel_size)
voxel_coords = np.array([voxel_grid.get_voxel(x) for x in np.asarray(pcd_down.points)])

slices = {}
for idx, coord in enumerate(voxel_coords):
    x = coord[0]
    if x not in slices:
        slices[x] = []
    slices[x].append((coord, pcd_down.points[idx]))

slices = {k: slices[k] for k in sorted(slices)}
```
to change the slice direction you need to adjust the coord[0] number, but i can't remember which was which.

I hope that helps

**Comment by saurabheights** [comment]
Try below code. The coordinate frame helps in visualizing X,Y,Z directions. Here I have used bounding box to crop the pointcloud but can also edit pointcloud points and colors field directly. Convert to numpy array and then back.

```
import open3d as o3d

lower_b = [0.0, 0.0, 0.0]
upper_b = [2.0, 2.0, 2.0]
bbox = o3d.geometry.AxisAlignedBoundingBox(min_bound=lower_b, max_bound=upper_b)
bbox.color = [1.0, 0.0, 0.0]

coordinate_frame = o3d.geometry.TriangleMesh.create_coordinate_frame(size=1)

ply_point_cloud = o3d.data.PLYPointCloud()
pcd = o3d.io.read_point_cloud(ply_point_cloud.path)

o3d.visualization.draw_geometries([bbox, pcd, coordinate_frame],
                                  zoom=0.3412,
                                  front=[0.4257, -0.2125, -0.8795],
                                  lookat=[2.6172, 2.0475, 1.532],
                                  up=[-0.0694, -0.9768, 0.2024])
pcd1 = pcd.crop(bbox)
o3d.visualization.draw_geometries([bbox, pcd1, coordinate_frame],
                                  zoom=0.3412,
                                  front=[0.4257, -0.2125, -0.8795],
                                  lookat=[2.6172, 2.0475, 1.532],
                                  up=[-0.0694, -0.9768, 0.2024])

```

**Comment by lurunawjan** [comment]
Thank you for the reply. Actually I'm trying to slice the 3D point cloud into 2D sectional area. Is it possible like that?

Thank you

**Comment by saurabheights** [comment]
Could you explain what you mean by slicing?

**Comment by lurunawjan** [comment]
Hello,

Actually I have a project on how to create a simple 3D model from 3D point cloud data. For this time, I use simple 3D point cloud object for example a box. Slicing means that I cut the 3D point cloud data in horizontal and vertical axis depend on the object. I'm still new with this and I hope anyone have idea to help me with this.

Thank you

**Comment by saurabheights** [comment]
@lurunawjan Well, my above comment should address your need, but for complex scenarious, you can directly work with pcd.points and pcd.colors.

PCD.points -> [convert to numpy array](http://www.open3d.org/docs/release/tutorial/geometry/working_with_numpy.html#From-open3d.PointCloud-to-NumPy) -> Do your transforms -> [convert to PCD.points](http://www.open3d.org/docs/release/tutorial/geometry/working_with_numpy.html#From-NumPy-to-open3d.PointCloud)

Just remember that you will have to do same operation for colors as well. If you drop half vertices in points array, you will have to drop half vertices in color array as well.

**Comment by theNded** [comment]
If you mean cropping, here is a minimal example:
https://github.com/isl-org/Open3D/blob/master/examples/python/geometry/point_cloud_crop.py,
and here is an interactive example:
https://github.com/isl-org/Open3D/blob/0f12276eb313456756425f3fbb00ad734bb88aed/examples/python/visualization/interactive_visualization.py#L15

**Comment by hhmaizi** [comment]
pcd = o3d.io.read_point_cloud(pcd_file_path) # your point cloud
pcd_roi = o3d.geometry.PointCloud() # point cloud of interest, I give an example extract points when its z coordinates is lower than 0.1.
points = np.asarray(pcd.points)
points_roi = points[points[:, 2] < 0.1] # using numpy array to slice your points
pcd_roi.points = o3d.utility.Vector3dVector(points_roi)
o3d.visualization.draw_geometries([pcd_roi], window_name='point cloud of interest')

**Comment by benjaminum** [comment]
There are multiple good answers. Marking this as solved
